### PR TITLE
[IA-4755] Forbid single quotes in params passed to SQL queries

### PR DESCRIPTION
**Pull request by mlilynolting**
__Jira ticket__: Child of https://broadworkbench.atlassian.net/browse/IA-4626

## Summary of changes

Quick and dirty patch to SQL hole in RuntimeV2Service: forbid single quotes in the label parameters' key-value pairs.

### What to test

- Try submitting requests with single quotes in label fields
- Try submitting requests in general to assert no regression issues

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

**Review by TomConner — CHANGES_REQUESTED** [review]
Bind variables would provide more protection. Can you look at using bind variables as suggested in this comment?

https://broadworkbench.atlassian.net/browse/IA-4626?focusedCommentId=85830

Happy to help sort this out!

**Comment by mlilynolting** [comment]
@TomConner I plan to convert this query to an appropriately parameterized one to make it truly secure. Wanted to get this messy fix up so we had the option of merging sooner and improving later.

**Review by TomConner — APPROVED** [review]
Approving this because it does provide some value. but hoping for bind variables ASAP.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeServiceInterp.scala [hunk 1]
```diff
@@ -262,6 +262,16 @@ class RuntimeServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
...
       creatorOnly <- F.fromEither(processCreatorOnlyParameter(userInfo.userEmail, params, ctx.traceId))
+
+      // prevent misbehaved user input, labelMap keys and values can not have a single quote
+      // keys and values are inserted directly into a SQL string which could cause sql injection
+      // story #IA-xxx will refactor the SQL string into Slick
+      _ <- F.raiseWhen(
+        labelMap.values.exists(value => value.contains("'")) || labelMap.keys.exists(key => key.contains("'"))
+      )(
+        BadRequestException(s"Invalid query parameter (single quote not allowed)", Some(ctx.traceId))
+      )
```
Comment (lucymcnatt) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeServiceInterp.scala:273]: This changes the behavior for runtimes, I think label maps are used in disks and apps as well, do we need to add similar logic to those services?

Comment (lucymcnatt) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeServiceInterp.scala:273]: that might be out of the scope of this particular ticket, but wanted to bring it up

**Review by cindy-broadinstitute — APPROVED** [review]

**Comment by mlilynolting** [comment]
jenkins retest

**Comment by mlilynolting** [comment]
Merging through known automation test flakes (RStudioSpec, RKernelSpec)

**Pull request merged**